posts/2024-07-12_teal_app_development_pharmaverseadam/teal-app-development.qmd

```diff
@@ -0,0 +1,208 @@
+    example_module(),
+    tm_data_table("Table View"),
+    tm_variable_browser("Variables")
+  )
+)
+```
+
+This is possible thanks to the magic that [`{teal}`](https://insightsengineering.github.io/teal/latest-tag/) is doing under the hood - passing the data object to each module. And now we will have access to a nice tabular view of the data, and a tool to explore each variable in greater detail.
+
+![](images/Asset 4.gif)\
+This is great on its own, but as a bonus, we even get the ability to build a report based on some of the modules that we have. For example, we can generate some plots in the variable browser, add them to the report and preview it. Some modules would also add a block of R code showing how to get the exact same data that was used to generate a report card.
+
+Finally, let’s add a simple barchart module that comes from the clinical modules package. We will use an example from the {teal.modules.clinical} documentation:
+
+``` r
+barchart_module <- tm_g_barchart_simple(
+  label = "ADAE Analysis",
+  x = data_extract_spec(
+    dataname = "ADAE",
+    select = select_spec(
+      choices = variable_choices(
+        pharmaverseadam::adae,
+        c("ARM", "ACTARM", "SEX")
+      ),
+      selected = "ACTARM",
+      multiple = FALSE
+    )
+  )
+)
+```
+
+![](images/Asset 5.gif)
+
+The best part about this module is that when a card is added to the report, it has R code that will reproduce exactly the same output that we see in the app.
+
+Here is the entire code for the application. In just 40 lines of code we were able to create a feature-rich application with the ability to interact with ADaM data, create visualizations and generate reproducible reports.
+
+``` r
+library(sparkline)
+library(teal)
+library(teal.data)
+library(teal.modules.clinical)
+library(teal.modules.general)
+
+data <- within(teal_data(), {
+  ADAE <- pharmaverseadam::adae
+  ADSL <- pharmaverseadam::adsl
+  # nolint end
+})
+datanames(data) <- c("ADAE", "ADSL")
+join_keys(data) <- default_cdisc_join_keys[datanames(data)]
+
+barchart_module <- tm_g_barchart_simple(
+  label = "ADAE Analysis",
+  x = data_extract_spec(
+    dataname = "ADAE",
+    select = select_spec(
+      choices = variable_choices(
+        pharmaverseadam::adae,
+        c("ARM", "ACTARM", "SEX")
+      ),
+      selected = "ACTARM",
+      multiple = FALSE
+    )
+  )
+)
+
+app <- init(
+  data = data,
+  modules = modules(
+    example_module(),
+    tm_data_table("Table View"),
+    tm_variable_browser("Variables"),
+    barchart_module
+  )
+)
+
+shinyApp(app$ui, app$server)
+```
+
+## **Conclusion** 
+
+In conclusion, [Teal](https://github.com/insightsengineering/teal) and [PharmaverseAdam](https://github.com/pharmaverse/pharmaverseadam) make it much easier to create interactive and reproducible clinical data dashboards. By following this guide, you can quickly build a Shiny app that not only visualizes your data but also maintains reproducibility and customization options. 
```

Comment:
`{pharmaverseadam}` should be all lowercase and within curly brackets - not sure if you want to match teal to that